### PR TITLE
[5.2] Added `onFlatFlat` method to collection

### PR DESCRIPTION
**Pull request by ninjaparade**
Adding `Drake` functionality to collections to allow calling `onFlatFlat` as per the lyrics in "For Free" Video  [here](https://www.youtube.com/watch?v=4I9bDV8VhcY)

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -288,7 +288,19 @@ public function flatten($depth = INF)
     {
         return new static(Arr::flatten($this->items, $depth));
     }
-
+    
+    /**
+     * Get a flattened array of the items in the collection.
+     *
+     * @param  int  $depth
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: int|INF

Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: INF is not an int

Comment (ninjaparade) [inline src/Illuminate/Support/Collection.php]: @GrahamCampbell Should that also be changed in the doc block for flatten?

Comment (GrahamCampbell) [inline src/Illuminate/Support/Collection.php]: yes please

**Comment by taylorotwell** [comment]
No tests so I can not really confirm if this is actually "flat flat". 

Rating: 1/10. Better luck next time.

**Comment by taylorotwell** [comment]
gov73zrtb9dyf8fv

**Comment by JosephSilber** [comment]
@taylorotwell if I convert it to TypeScript, will you merge it?

**Comment by GrahamCampbell** [comment]
@taylorotwell YES!

**Comment by GrahamCampbell** [comment]
For those looking for the source of that brillinant meme, look no further than https://hackathon.starbs.net/ with the source at https://github.com/YorkZamzar2016.

**Pull request closed without merging**